src/telemetry/filters.ts

```diff
@@ -0,0 +1,74 @@
+import { ITelemetryItem } from '@microsoft/applicationinsights-web';
+import {
+  DEVX_API_URL,
+  GRAPH_API_SANDBOX_URL,
+  GRAPH_URL,
+} from '../app/services/graph-constants';
+import {
+  sanitizeGraphAPISandboxUrl,
+  sanitizeQueryUrl,
+} from '../app/utils/query-url-sanitization';
+
+export function filterTelemetryTypes(envelope: ITelemetryItem) {
+  const baseType = envelope.baseType || '';
+  const telemetryTypesToInclude = [
+    'EventData',
+    'MetricData',
+    'PageviewData',
+    'ExceptionData',
+    'RemoteDependencyData',
+  ];
+  return telemetryTypesToInclude.includes(baseType);
+}
+
+export function filterRemoteDependencyData(envelope: ITelemetryItem): boolean {
+  const baseData = envelope.baseData || {};
+  if (envelope.baseType !== 'RemoteDependencyData') {
+    return true;
+  }
+
+  const targetsToInclude = [GRAPH_URL, DEVX_API_URL, GRAPH_API_SANDBOX_URL];
+  const urlObject = new URL(baseData.target || '');
+  if (!targetsToInclude.includes(urlObject.origin)) {
+    return false;
+  }
```

Comment:
It could be more likely for the `urlObject.origin` to contain the base URLs than the other way round.

Reply:
The aim is to compare base URLs on both sides